lib/sshutils/sftp/sftp_test.go

```diff
@@ -353,208 +410,23 @@ func TestUpload(t *testing.T) {
 					createFile(t, filepath.Join(tempDir, file))
 				}
 			}
-			for i := range tt.srcPaths {
-				tt.srcPaths[i] = filepath.Join(tempDir, tt.srcPaths[i])
-			}
-			for i := range tt.globbedSrcPaths {
-				tt.globbedSrcPaths[i] = filepath.Join(tempDir, tt.globbedSrcPaths[i])
-			}
-			tt.dstPath = filepath.Join(tempDir, tt.dstPath)
-
-			cfg, err := CreateUploadConfig(tt.srcPaths, tt.dstPath, tt.opts)
-			require.NoError(t, err)
-			// use all local filesystems to avoid SSH overhead
-			cfg.dstFS = &localFS{}
-			err = cfg.initFS(nil)
-			require.NoError(t, err)
-
-			ctx := context.Background()
-			err = cfg.transfer(ctx)
-			if tt.errCheck == nil {
-				require.NoError(t, err)
-				srcPaths := tt.srcPaths
-				if len(tt.globbedSrcPaths) != 0 {
-					srcPaths = tt.globbedSrcPaths
-				}
-				checkTransfer(t, tt.opts.PreserveAttrs, tt.dstPath, srcPaths...)
-			} else {
-				tt.errCheck(t, err, tempDir)
-			}
-		})
-	}
-}
-
-func TestDownload(t *testing.T) {
```

Comment:
Any reason you deleted this whole test? You should at least work the test cases into `TestTransferFiles` as they cover code not covered in that test IIRC

Reply:
I see no reason for both TestUpload and TestDownload when we barely differentiate between uploads and downloads (especially in the tests where we use a local fs for both anyway). I copied over one slightly different test case, but all the others looked identical to me.

Reply:
Can you double check with `go test -cover` and friends, running package tests with `TestDownload` removed and then again with it included? If `TestDownload` doesn't add any coverage or it can be done just by adding more subtests to `TestTransferFiles` that's fine, but I think sure the download and upload tests were necessary for better coverage when I originally wrote the code at least. Def fact check me, best case the code isn't needed and the test code can be slimmed down

Reply:
Checked, only file that lost coverage was parse.go due to adding `ParseSources`, which I've added a test for.